### PR TITLE
Rahulyadav

### PR DESCRIPTION
**Pull request by rahulyadav170923**
sir i have made some changes according to previous review .
a). used qsTr in some areas
b).removed previous warnings
c). added border to droppable zones
d).added level changing functionality
d).structured the code.
i still have two problems. error(QML Rectangle: Cannot anchor to a null item.) and other one is that the operator does not go back to original position.i am working on sizing the elements . iam also working on a different structure for activity(making components) so that i can add multiple rows.

### CODE REVIEW

File: src/activities/guesscount/ActivityInfo.qml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+/* GCompris - ActivityInfo.qml
+ *
+ * Copyright (C) 2015 Your Name <yy@zz.org>
```
Comment (petitlapin) [inline src/activities/guesscount/ActivityInfo.qml]: can you change the copyright (2016 and your name/email)

File: src/activities/guesscount/ActivityInfo.qml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  demo: true
+  title: qsTr("Guessing the algebraic expression for the answer")
+  description:qsTr("guess the algebraic expression and click on the buttons to add them.")
+  //intro: "guesscount activity"
+  goal: qsTr("To teach algebra")
+  prerequisite: qsTr("knowledge of arithematic oprations")
+  manual: ""
+  credit: ""
+  section: "math"
+  createdInVersion:1
```
Comment (petitlapin) [inline src/activities/guesscount/ActivityInfo.qml]: you are not supposed to change this attribute :). It corresponds to the GCompris version you create the activity, it's not a versioner for activity changes.

File: src/activities/guesscount/Guesscount.qml [hunk 3]
```diff
@@ -0,0 +1,462 @@
+/* GCompris - guesscount.qml
+ *
+ * Copyright (C) 2015 YOUR NAME <xx@yy.org>
```
Comment (petitlapin) [inline src/activities/guesscount/Guesscount.qml]: can you change the copyright (2016 and your name/email)

File: src/activities/guesscount/Guesscount.qml [hunk 4]
```diff
@@ -0,0 +1,462 @@
...
+ *
+ * Authors:
+ *   <THE GTK VERSION AUTHOR> (GTK+ version)
```
Comment (petitlapin) [inline src/activities/guesscount/Guesscount.qml]: update here and the following line also

File: src/activities/guesscount/Guesscount.qml [hunk 5]
```diff
@@ -0,0 +1,462 @@
...
+                                console.log(row4.children[0].children[1].obj.datavalue*row4.children[2].children[1].obj.datavalue);
+                                dragTarget.calculated_value=row4.children[0].children[1].obj.datavalue*row4.children[2].children[1].obj.datavalue
+
+                            }
+                            row4.children[5].children[0].text=Number(dragTarget.calculated_value).toString()
+                            if(dragTarget.calculated_value==items.guesscount)
+                            {
+
+                                row4.children[5].children[0].text=Number(dragTarget.calculated_value).toString()
+                                if(items.question_no==3){
```
Comment (petitlapin) [inline src/activities/guesscount/Guesscount.qml]: can you uniform the coding style, after the if(), there is a space and the accolade:
if(test) { 
   ...
}

File: src/activities/guesscount/Guesscount.qml [hunk 6]
```diff
@@ -0,0 +1,462 @@
...
+                id:result
+                width:100
+                height: 100
+                color:"white"
+                border.color: "black"
+                GCText{
+                    anchors.horizontalCenter: parent.horizontalCenter
+                    anchors.verticalCenter: parent.verticalCenter
+                    fontSize: mediumSize
+                    text:Number(0).toString()
```
Comment (petitlapin) [inline src/activities/guesscount/Guesscount.qml]: "0" directly?

File: src/activities/guesscount/guesscount.js [hunk 7]
```diff
@@ -0,0 +1,112 @@
+/* GCompris - guesscount.js
+ *
+ * Copyright (C) 2015 YOUR NAME <xx@yy.org>
```
Comment (petitlapin) [inline src/activities/guesscount/guesscount.js]: copyright

File: src/activities/guesscount/Guesscount.qml [hunk 8]
```diff
@@ -0,0 +1,462 @@
...
+            spacing: 50
+            anchors{
+                top:row3.bottom
+                topMargin:10
+                left:parent.left
+                leftMargin: 300
+            }
+            Repeater{
+                id:repeat
+                model:3
```
Comment (petitlapin) [inline src/activities/guesscount/Guesscount.qml]: how you do handle levels with more than 2 numbers and 1 operator?
I mean, for example, you have: 1 2 3 and need to find 6. A solution would be 1 + 2 + 3, is it handle?

**Comment by petitlapin** [comment]
Hi (it's JohnnyJ from the chat),

I haven't tested it but here are a few remarks.

JohnnyJ

**Comment by rahulyadav170923** [comment]
yes johnny i'll make the changes in next commmit by the end of the day.

**Comment by rahulyadav170923** [comment]
regarding handling levels .i have made changes that will appear in the
upcoming commit.but i havent made it such that it can handle 1+2+3.i have
made it similar to gtk version with some changes .to fulfill the the
objective that you mentioned i will be making a practice mode where 1+2+3
 or other such problems can be solved.please let me know how you feel about
the practice mode.i mentioned it to <bdoin> sir.

**Pull request closed without merging**